test_browser_automation.py

```
@@ -0,0 +1,431 @@
+        self.assertTrue(path.endswith(".png"), f"Expected .png fallback, got: {path}")
+
+    def test_no_print_on_format_detection_exception(self):
+        """No output is printed when the format detection block raises."""
+        unknown_bytes = b"\x00\x01\x02\x03" * 16
+        with patch("browser_automation.Image") as mock_img_mod, \
+             patch("builtins.print") as mock_print:
+            mock_img_mod.open.side_effect = Exception("unrecognised format")
+            try:
+                _run_screenshot_with_bytes(self.ba, unknown_bytes)
+            except Exception:
+                pass
```

Ruff (0.15.12)

[error] `try`-`except`-`pass` detected, consider logging the exception

(S110)